### ISSUE TITLE
Docker Swarm Future Concerns!!! [Question]

### ISSUE THREAD

**Issue by mehdiamenein** [caprover/caprover#908]
Hi everyone, 

First of all, thank you so much for this amazing tool, I am a Junior DevOp as well as a Junior-mid experience Full-stack developer (~ 5 years so long way to be a senior). For our company we are using Azure DevOp and for my personal works I was using heroku free dyno for prototyping versions and CircleCI + Digital Ocean for personal or production projects.

I dockerize all my projects cause it is generally easier to deploy them, however still a full CI/CD pipeline can take me between 3 to 10 hours and I am never 100% satisfied with the outcome.

I saw this tool and was just "WOW", I immediately made a test droplet and when for a CapRover instance, then I saw the clustering and read the shell scripts and got a bit hesitated...

This tool is using docker swarm and there are a lot of discussions around this container orchestration platform, some articles on it:
https://medium.com/@markuman/is-docker-swarm-mode-eol-7a3f316116a3
https://www.optimum-web.com/is-docker-swarm-dead-is-anyone-using-swarm-in-production-in-2019-the-future-of-docker-swarm/
https://www.reddit.com/r/docker/comments/es1kxr/future_of_docker_swarm/
https://devclass.com/2020/02/25/mirantis-to-keep-docker-swarm-buzzing-around-pledges-new-features/
https://thenewstack.io/kubernetes-vs-docker-swarm-whats-the-difference/

so long story short there are a lot of mixed discussions on docker swarm technology. now my question is:

"Do you have any general plans or did you anticipated the **potential** Swarms downfall in next couple of years"

**Comment by githubsaturn** [comment]
https://github.com/caprover/caprover/issues/722#issuecomment-650439019
